### PR TITLE
[LER-103] - integrate search screen

### PR DESCRIPTION
**Pull request by valenchus**

### CODE REVIEW

File: src/app/(auth)/(tabs)/explore/searchScreen/index.tsx [hunk 1]
```diff
@@ -1,45 +1,114 @@
...
+      });
+    } else {
+      setParams(() => {
+        const newParams = { filter: quickFilterSelected.type };
+        return newParams;
+      });
+    }
+  }, [quickFilterSelected.type]);
+
   useEffect(() => {
```
Comment (fabrizioserial) [inline src/app/(auth)/(tabs)/explore/searchScreen/index.tsx:92]: Remove this useEffect

File: src/app/(auth)/(tabs)/explore/utils.ts [hunk 2]
```diff
@@ -256,6 +256,206 @@ export const mockedSearchResults = [
     status: 'locked' as SearchResultStatus,
   },
 ];
+export const mockedSearchResultsWithPagination = [
```
Comment (fabrizioserial) [inline src/app/(auth)/(tabs)/explore/utils.ts:259]: Remove this mock data

File: src/redux/service/home.service.ts [hunk 3]
```diff
@@ -1,15 +1,23 @@
 import { api } from './api';
+import { SearchQueryParams, SearchResponse } from './types/home.response';
 
 export const homeApi = api.injectEndpoints({
   endpoints: (builder) => ({
     programEnrolled: builder.query({
       query: () => ({ url: '/home', method: 'GET' }),
     }),
+    search: builder.query<SearchResponse, { page?: string; filter?: string; search?: string }>({
```
Comment (fabrizioserial) [inline src/redux/service/home.service.ts:9]: If we are using SearchQueryParams in the next line, why are we not using that here too? 
`<SearchResponse, { page?: string; filter?: string; search?: string }>` => `<SearchResponse, SearchQueryParams >`

File: src/app/(auth)/(tabs)/explore/searchScreen/index.tsx [hunk 4]
```diff
@@ -97,38 +169,29 @@ const SearchScreen = () => {
...
+              title={data.item.name}
+              description={data.item.description}
+              status={data?.item.status}
+              progress={data?.item.progress}
+              imgUrl={data?.item.imgUrl}
+            />
+          )}
+        />
+      ) : (
+        <ErrorDisplay type="no-results" />
```
Comment (schcolnikj) [inline src/app/(auth)/(tabs)/explore/searchScreen/index.tsx:193]: image
No esta centrado

File: src/app/(auth)/(tabs)/explore/searchScreen/index.tsx [hunk 5]
```diff
@@ -97,38 +169,29 @@ const SearchScreen = () => {
...
-            }}
-          >
-            <ErrorDisplay type="no-results" />
-          </StyledColumn>
-        )}
-      </ScrollView>
+      {isLoading ? (
+        <SearchScreenSkeleton />
+      ) : resultsToShow && resultsToShow?.length > 0 ? (
+        <FlatList
```
Comment (schcolnikj) [inline src/app/(auth)/(tabs)/explore/searchScreen/index.tsx:175]: Habría que esconder el scroll.

**Review by schcolnikj — APPROVED** [review]

**Pull request merged**